src/mqt/bench/benchmark_generation.py

```diff
@@ -90,10 +90,10 @@ def generate_filename(
     base = f"{benchmark_name}_{level}"
 
     if level == "nativegates" and gateset is not None and opt_level is not None:
-        return f"{base}_{gateset.name}_opt{opt_level}_{num_qubits}"
+        return f"{base}_{gateset.description}_opt{opt_level}_{num_qubits}"
```

Comment:
I am fairly sure the description needs some kind of sanitization.
Many descriptions of targets in the wild will use spaces (`"Hardware Provider XZY Target"`). 
Based on the above comment, the gateset should be a serialized/stringified version of the gateset and the device description should be a sanitized version of the description.